schematic/schemas/json_schema_component_generator.py

```diff
@@ -258,51 +254,15 @@ def get_component_json_schema(
             inputMModelLocationType="local",
             data_model_labels="class_label",
         )
-        data_model_json_schema = DataModelJSONSchema(
+        json_schema_generator = JSONSchemaGenerator(
             jsonld_path=metadata_model.inputMModelLocation,
             graph=metadata_model.graph_data_model,
         )
 
-        self.incomplete_component_json_schema = (
-            data_model_json_schema.get_json_validation_schema(
-                source_node=self.component, schema_name=schema_name
-            )
+        self.component_json_schema = json_schema_generator.create_json_schema(
+            datatype=self.component, schema_name=schema_name
         )
```

Comment:
The use of creating class instances is a bit puzzling to me. Generally, I'd only recommend using classes when the focus of the class is tracking the state of data, and interacting with that data in some way. The way that this is used, `create_json_schema` doesn't need to be a part of a `JSONSchemaGenerator` class, it just needs to be some function that you can invoke and get a response from.

To reinforce my thought on this, `JSONSchemaGenerator` is created 2 lines above, used here, and then left for garbage collection.

Reply:
@BryanFauble If I had more time, I think `JSONSchemaGenerator` would have eventually gotten refactored into a function. I was partially there by moving most of state tracking into `NodeProcessor`. There will be a follow-up ticket to finish doing this.